blockchain/blockdao/blockdao.go

```diff
@@ -120,6 +124,7 @@ type (
 		compressBlock bool
 		kvstore       db.KVStore
 		indexer       BlockIndexer
+		sf            factory.Factory
```

Comment:
sf is just another indexer, don't use type factory.Factory

Reply:
you mean BlockIndexer interface?sf need change and add PutBlock,DeleteTipBlock,factory.Factory cannot use block.Block